src/zap.ts

```diff
@@ -0,0 +1,55 @@
+import { ChainId, isEthereum, isFantom } from "./chain";
+import { EthAddress, ZeroAddress } from "./helpers";
+import { Token, VaultMetadataOverrides } from "./types";
+
+export type ZapInWith = keyof Token["supported"];
+
+type ZapInDetails = { isZapInSupported: boolean; zapInWith?: ZapInWith };
+
+type ZapInArgs = {
+  chainId: ChainId;
+  token?: Partial<Token>;
+  vaultMetadata: VaultMetadataOverrides | null;
+};
+
+export function getZapInDetails({ chainId, token, vaultMetadata }: ZapInArgs): ZapInDetails {
+  if (!token?.supported || !vaultMetadata?.zapInWith) {
+    return { isZapInSupported: false };
+  }
+
+  if (isEthereum(chainId)) {
+    if (token.address === EthAddress) {
+      return { isZapInSupported: Boolean(token.supported.zapper), zapInWith: "zapperZapIn" };
+    }
+
+    const { zapInWith } = vaultMetadata;
+
+    if (zapInWith && isValidZap(zapInWith, token.supported)) {
+      const isZapInSupported = Boolean(token.supported.zapper) && Boolean(token.supported[zapInWith]);
+      return { isZapInSupported, zapInWith };
+    }
+  }
+
+  if (isFantom(chainId)) {
+    const zapInWith = "ftmApeZap"; // Hardcoded for now
+
+    if (token.address === ZeroAddress) {
+      return { isZapInSupported: Boolean(token.supported[zapInWith]), zapInWith };
+    }
```

Comment:
Also we could remove this part of the code since it can be treated as any token.

Reply:
Done with https://github.com/yearn/yearn-sdk/pull/280/commits/8cf71360739164ae605fed430332d39471b02bdd